definitions/marts/looker_studio_marts/ecf/ls_ecf_correct_cohort_checker.sqlx

```diff
@@ -0,0 +1,241 @@
+        start_date_suggestion_match: "1 or 0 if the suggested cohort based on induction start date matches with the current cohort in the servce.",
+        earliest_dec_suggestion_match: "1 or 0 if the suggest cohort based on the earliest funded declaration for the participant matches the current cohort in the service.",
+        has_relevant_declarations: "TRUE/FALSE if the participant has any s/e/p/p declarations present.",
+        cohort_rolled_over_21_24: "TRUE/FALSE flag from the service as to whether the participant's cohort has been changed from 2021 to 2024.",
+        has_induction_start_date: "TRUE/FALSE if the induction_start_date field is not null."
+    }
+}
+
+/*** ECF Cohort Corrections Looker Studio Data Mart ***/
+
+/*
+  Objective:
+    To gather all ECTs & Mentors inducted within 2021-2023 cohorts for evaluation as to whether they are allocated to correct the cohort based on criteria below.
+
+  Instructions:
+    For ECTs in the "wrong" cohort with declarations:
+      WHERE
+        eligible_for_funding = TRUE
+        AND
+        participant_role = 'ECT'
+        AND 
+        declaration_id IS NOT NULL
+        AND
+        niot_partnership_chk = 0
+        AND
+        start_date_suggestion_match = 0
+      
+    For Mentors in the "wrong" cohort with declarations:
+      WHERE
+        eligible_for_funding = TRUE
+        AND
+        participant_role = 'Mentor'
+        AND 
+        declaration_id IS NOT NULL
+        AND
+        niot_partnership_chk = 0
+        AND
+        earliest_dec_suggestion_match = 0
+
+    For ECTs in the "wrong" cohort without declarations:
+      WHERE
+        eligible_for_funding = TRUE
+        AND
+        participant_role = 'ECT'
+        AND 
+        declaration_id IS NULL
+        AND
+        has_relevant_declarations = FALSE
+        AND
+        niot_partnership_chk = 0
+        AND
+        start_date_suggestion_match = 0
+
+  Dependencies:
+    dataform.ecf_declarations
+*/
+
+-- Gather ECF Declarations
+WITH declarations AS( 
+  SELECT
+     -- Manually assign the user_id for specific participant profile where the profile no longer exists in latest_cpd data and has no superceeded record to call upon.
+     CASE 
+      WHEN dec.participant_profile_id = '8cd0928c-ecad-482f-95cd-6a0ff3e840ad' THEN '79dee4de-a6dd-407c-ab1c-690fcb6d0056'
+      ELSE dec.user_id
+     END AS user_id
+    ,dec.participant_profile_id
+    ,dec.participant_course AS participant_role
+    ,dec.schedule_identifier
+    ,dec.induction_start_date AS induction_start_date
+    ,dec.cohort
+    ,dec.eligible_for_funding
+    ,dec.partnership_id
+    ,dec.partnership_relationship
+    ,dec.lead_provider_name AS induction_lead_provider_name
+    ,dec.cohort_rolled_over_21_24
+    ,IF(dec.state IN ('submitted', 'paid', 'payable', 'eligible'),dec.declaration_id, NULL) AS declaration_id

```

Comment:
Could you just use the field you created in the subquery to do this check?

Reply:
I'm also a bit uneasy about this logic because that's not the value of the fields as reported by lead providers. We're just overwriting the values with nulls and naming the fields the same thing. Why not just filter out any records where declarations have been received that don't meet the criteria? You'd still have the records where no declarations have been received and the records where the correct declarations have been received.

Reply:
Otherwise you just end up with a lot of blank records in the output table with no way to distinguish between them.